flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -137,7 +162,16 @@ else if (remaining == 0) {
 		// spanning record case
 		if (this.spanningWrapper.hasFullRecord()) {
 			// get the full record
-			target.read(this.spanningWrapper.getInputView());
+			try {
+				target.read(this.spanningWrapper.getInputView());
+			} catch (EOFException e) {
+				Optional<String> deserializationError = this.spanningWrapper.getDeserializationError(1);
```

Comment:
I do not quite understand why we set `addToReadBytes` as 1 here.

If the `target.read` is successful, then we do `spanningWrapper.getDeserializationError(0)` in the following `moveRemainderToNonSpanningDeserializer` and it makes sense, otherwise we do `spanningWrapper.getDeserializationError(1)`.

The `spanningWrapper.getDeserializationError(0)` may be also suitable for the exception case? Because we only want to show some internal informations during exceptions for debugging. Then we just need one check.

Reply:
For the first part, see above.

Regarding the number of checks: basically, the `spanningWrapper.getDeserializationError(1)` check is for the case where we tried to consume more bytes than available, `spanningWrapper.getDeserializationError(0)` for the case where we did not consume all bytes we were supposed to consume. Only one of the two checks will be executed in any case.

-> I hope, this will become clearer after the refactoring.